ghcide/src/Development/IDE/Core/Shake.hs

```diff
@@ -1125,6 +1128,17 @@ defineEarlyCutOffNoFile recorder f = defineEarlyCutoff recorder $ RuleNoDiagnost
     if file == emptyFilePath then do (hash, res) <- f k; return (Just hash, Just res) else
         fail $ "Rule " ++ show k ++ " should always be called with the empty string for a file"
 
+data SourceFileOrigin = FromProject | FromDependency
+
+getSourceFileOrigin :: NormalizedFilePath -> SourceFileOrigin
+getSourceFileOrigin f =
+    case [".hls", "dependencies"] `isInfixOf` (splitDirectories file) of
```

Comment:
@wz1000  What do you think of this? It still uses `isInfixOf` but not show. I don't see any predicate in `System.FilePath` that is like `isInfixOf` but for directories being contained in a `FilePath`.

Reply:
This seems alright to me